### PR TITLE
add oauth2 test, reorganize authorization tests

### PR DESCRIPTION
**Pull request by iscai-msft**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: .changeset/strong-goats-pump.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@azure-tools/cadl-ranch-specs": minor
```
Comment (timotheeguerin) [inline .changeset/strong-goats-pump.md:2]: Do we want to start bunping minor? Maybe just patch for now still

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -859,8 +859,8 @@ packages:
       '@cspell/dict-docker': 1.1.1
       '@cspell/dict-dotnet': 2.0.1
       '@cspell/dict-elixir': 2.0.1
-      '@cspell/dict-en-gb': 1.1.33
       '@cspell/dict-en_us': 2.3.3
+      '@cspell/dict-en-gb': 1.1.33
```
Comment (timotheeguerin) [inline pnpm-lock.yaml:863]: Weird change

Comment (timotheeguerin) [reply pnpm-lock.yaml:863]: Hopefully it doesn’t keep bouncing order

**Review by timotheeguerin — APPROVED** [review]

File: .changeset/strong-goats-pump.md (hunk 1 above)
Comment (iscai-msft) [reply .changeset/strong-goats-pump.md:2]: oh good point, let me fix that

File: pnpm-lock.yaml (hunk 2 above)
Comment (iscai-msft) [reply pnpm-lock.yaml:863]: yeah idk what happened, we shall see

**Pull request merged**